app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java

```diff
@@ -465,13 +468,18 @@ public static void disablePullNotifications() {
 
     public static void clearNotificationsForActiveAccount(@NonNull Context context, @NonNull AccountManager accountManager) {
         AccountEntity account = accountManager.getActiveAccount();
-        if (account != null) {
-            account.setActiveNotifications("[]");
-            accountManager.saveAccount(account);
```

Comment:
Database access! On the main thread! And in 90% of cases it's redundant!

Reply:
Could it be caught by `android.os.StrictMode`? Could you try it out please? Event `StrictMode.enableDefaults()` should be enough.

Reply:
THIS is what causing lags for me! I thought it's some debugging in my branch but it's *this*. Good job finding it!